### ISSUE TITLE
how long it would take to run RunUMAP ?

### ISSUE THREAD

**Issue by zqyhyunbin** [satijalab/seurat#1467]
hello, i am following the tutorial here with my own data:[https://satijalab.org/seurat/v3.0/immune_alignment.html](url)

and i use less than 1000 cell to test it, but when it getting to the RunUMAP step, it takes really long time, more that 7hr, and it's still running, is that sth wrong or it's the natural?

the feedback i got from RunUMAP is like:
UMAP(a=None, angular_rp_forest=False, b=None, init='spectral',
   learning_rate=1.0, local_connectivity=1, metric='correlation',
   metric_kwds=None, min_dist=0.3, n_components=2, n_epochs=None,
   n_neighbors=30, negative_sample_rate=5, random_state=None,
   repulsion_strength=1.0, set_op_mix_ratio=1.0, spread=1.0,
   target_metric='categorical', target_metric_kwds=None,
   target_n_neighbors=-1, target_weight=0.5, transform_queue_size=4.0,
   transform_seed=42, verbose=True)
Construct fuzzy simplicial set
.
.
.

**Comment by SeadonXing** [comment]
For less than 1000 cells will finish immediately for me. There must be something wrong for you. You may try to run the example to check: 
pbmc_small <- RunUMAP(object = pbmc_small, dims = 1:5);
DimPlot(object = pbmc_small, reduction = 'umap');
If there still no output, I suggest reinstall umap-learn.

Best

**Comment by zqyhyunbin** [comment]
Many thanks, there must sth wrong with UMAP
